### ISSUE TITLE
Add filtering capabilities

### ISSUE THREAD

**Issue by Sermus** [tomgi/git_stats#13]
It would be awersome to have some filtering capabilities while gathering the stats:
1) Ignore certain files by names or extensions
2) Ignore certain paths
3) Ignore ranges of revisions

For example svnstat provided quite an elegant solution to 1 and 2 by giving an opportunity to set ignore list with apache-like wildcards. Motivation for such filtering is that some files in repository could be out of interest (autogenerated, imported 3rdParty data and so on)

**Comment by BrunoSpy** [comment]
:+1:

**Comment by nsams** [comment]
as a workaround I did filter-branch on the repo before running git_stats:

    git filter-branch --index-filter 'git rm --cached --ignore-unmatch -r *.svg' HEAD

**Comment by paulz** [comment]
:+1:

**Comment by trentpolack** [comment]
Yeah, I'd adore this too. This is the only repo analyzer that's actually quite wonderful, but my game project has oodles of binary files that are getting included on top of everything else (and the binary files are ~120gb, so moving them around or temporarily removing them seems... Worrisome).
